sherpa/plot/bokeh_backend.py

```diff
@@ -0,0 +1,829 @@
+            axes.yaxis.axis_label = ylabel
+
+    @add_kwargs_to_doc(kwargs_doc)
+    @translate_args
+    def histo(self, xlo, xhi, y, *,
+              yerr=None, title=None,
+              xlabel=None, ylabel=None,
+              overplot=False, clearwindow=True,
+              xerrorbars=False,
+              yerrorbars=False,
+              ecolor=None,
+              capsize=None,
+              xlog=False,
+              ylog=False,
+              linestyle='None',
+              drawstyle='default',
+              color=None,
+              alpha=None,
+              marker='None',
+              markerfacecolor=None,
+              markersize=None,
+              label=None,
+              linewidth=None,
+              linecolor=None,
+              ):
+        """Draw histogram data.
+
+        The histogram is drawn as horizontal lines connecting the
+        start and end points of each bin, with vertical lines connecting
+        consecutive bins. Non-consecutive bins are drawn with a
+        (NaN, NaN) between them so no line is drawn connecting them.
+
+        Points are drawn at the middle of the bin, along with any
+        error values.
+
+        Note that the linecolor is not used, and is only included
+        to support old code that may have set this option (use `color` instead).
+
+        Parameters
+        ----------
+        x0 : array-like or scalar number
+            lower bin boundary values
+        x1 : array-like or scalar number
+            upper bin boundary values
+        y : array-like or scalar number
+            y values, same dimension as `x0`.
+        {kwargs}
+        """
+        if linecolor is not None:
+            logger.warning("The linecolor attribute ({}) is unused.".format(linecolor))
+
+        x, y2 = histogram_line(xlo, xhi, y)
+        # Note: this handles clearing the plot if needed.
+        #
+        objs = self.plot(x, y2, yerr=None, xerr=None,
+                title=title, xlabel=xlabel, ylabel=ylabel,
+                overplot=overplot, clearwindow=clearwindow,
+                xerrorbars=False, yerrorbars=False,
+                ecolor=ecolor, capsize=capsize,
+                xlog=xlog, ylog=ylog,
+                linestyle=linestyle,
+                linewidth=linewidth,
+                drawstyle=drawstyle,
+                color=color, marker=None, alpha=alpha,
+                xaxis=False, ratioline=False)
+
+        # Draw points and error bars at the mid-point of the bins.
+        # Use the color used for the data plot: should it
+        # also be used for marker[face]color and ecolor?
+        #
+        xmid = 0.5 * (xlo + xhi)
+        xerr = (xhi - xlo) / 2 if xerrorbars else None
+        yerr = yerr if yerrorbars else None
+
+        try:
+            color = objs[0].get_color()
+        except (AttributeError, IndexError):
+            pass
+
+        # Do not draw a line connecting the points
+        #
+        # Unlike plot, using errorbar for both cases.
```

Comment:
Is this comment still true (it's true for matplotlib but I don't think is a requirement for al backends but I have not really paged in all this code)?